code/modules/projectiles/updated_projectiles/guns/energy.dm

```diff
@@ -365,3 +365,46 @@
 	update_icon(user)
 
 	return TRUE
+
+//-------------------------------------------------------
+//Deathsquad-only gun -- Model 2419 pulse rifle, the M19C4.
+
+/obj/item/weapon/gun/energy/lasgun/pulse
+	name = "\improper M19C4 pulse energy rifle"
+	desc = "A heavy-duty, multifaceted energy weapon that uses pulse-based beam generation technology to emit powerful laser blasts. Because of its complexity and cost, it is rarely seen in use except by specialists and front-line combat personnel. This is a testing model issued only for Asset Protection."
+	force = 23 //Slightly more heftier than the M43, but without the stock.
+	icon_state = "m19c4"
+	item_state = "m19c4"
+	max_shots = 100//codex stuff
+	load_method = CELL //codex stuff
+	ammo = /datum/ammo/energy/lasgun/pulse
```

Comment:
this is undefined

Reply:
Added an inhand sprite, as well as ammo fix.